File: CSharp/DependencyBreakingKatas/SubclassAndOverride/MarketingCampaign.cs
using System;

namespace Org.Codecop.Dependencies.SubclassAndOverride
{
    public class MarketingCampaign
    {

        private delegate long TimeSupplier();

        private readonly TimeSupplier SystemTimeMillis;

        public MarketingCampaign()
        {
            this.SystemTimeMillis = () => (long)DateTime.Now.TimeOfDay.TotalMilliseconds;
        }

        public bool IsActive()
        {
            return MilliSeconds() % 2 == 0;
        }

        private long MilliSeconds()
        {
            return this.SystemTimeMillis();
        }

        public bool IsCrazySalesDay()
        {
            return DayOfTheWeek().Equals(DayOfWeek.Friday); ;
        }

        private DayOfWeek DayOfTheWeek()
        {
            return DateTime.Now.DayOfWeek;
        }
    }
}

File: PHP/src/B/MarketingCampaign.php
<?php

namespace Org\Codecop\Dependencies\B;

class MarketingCampaign {

    public function isActive(): bool {
        return $this->milliSeconds() % 2 == 0;
    }

    private function milliSeconds() {
        return floor(microtime(true) * 1000);
    }

    public function isCrazySalesDay(): bool {
        return $this->dayOfWeek() == 4; // Friday
    }

    private function dayOfWeek() {
        return date('N');
    }
}
